src/elements/chatbox/chatbox.js

```diff
@@ -0,0 +1,60 @@
+import './chatbox.less'
+import React, { useState, useEffect } from 'react'
+import url from '/util/url'
+
+export const ChatBox = ({ roomId, ...props }) => {
+  const [messageInput, setMessageInput] = useState('')
+  const [chatLog, setChatLog] = useState('')
+  const [chatSocket, setChatSocket] = useState()
+  // const socketRef = useRef()
+
+  const handleSend = () => {
+    // socketRef.current.send(messageInput)
+    const toSend = JSON.stringify({ message: messageInput })
+    chatSocket.send(toSend)
+    setMessageInput('')
+  }
+
+  const handleChange = (e) => {
+    setMessageInput(e.target.value)
+  }
+
+  useEffect(() => {
+    console.log('Connecting new websocket to ' + url('api_ws.chatSocket', { args: { id: roomId } }))
+    const newSocket = new window.WebSocket(
+      url('api_ws.chatSocket', { args: { id: roomId } })
+    )
+
+    newSocket.onmessage = (e) => {
+      const data = JSON.parse(e.data)
+      setChatLog((previous) => previous + data.message + '\n')
+    }
+
+    newSocket.onclose = (e) => {
+      console.error('Chat socket closed unexpectedly')
+    }
+    setChatSocket(newSocket)
+    // socketRef.current = newSocket
+    // return () => {
+    //   console.log('Socket cleanup?')
+    // }
```

Comment:
Definitely on the right track here: a useEffect with an empty array of dependencies will be called when the component is mounted (and not on re-renders), any returned function will be called when the component is unmounted, which like you have commented out here, is the perfect time to cleanup the WebSocket reference.

Otherwise, you could end up with  extraneous websocket event handlers that will be calling useState references on a component that is no longer mounted.